### PR TITLE
Refactor Deeply Read Cache

### PR DESCRIPTION
**Pull request by nitro-marky**
## What does this change?

This refactors the frontend nextgen api which delivers the Deeply Read data for the next engagement test.

Before:

Screenshot 2021-01-07 at 16 36 05

After:
Screenshot 2021-01-11 at 10 51 48

### CODE REVIEW

**Comment by nitro-marky** [comment]
Shoutout to @jfsoul for some sterling work!

**Review by shtukas — APPROVED** [review]

File: onward/app/feed/DeeplyReadAgent.scala [hunk 1]
```diff
@@ -75,26 +75,23 @@ class DeeplyReadAgent(contentApiClient: ContentApiClient, ophanApi: OphanApi) ex
...
-  def refresh()(implicit ec: ExecutionContext): Future[Unit] = {
+  def refresh()(implicit ec: ExecutionContext): Future[Seq[Option[(OphanDeeplyReadItem, Content)]]] = {
     log.info(s"[cb01a845] Deeply Read Agent refresh()")
     /*
         Here we simply go through the OphanDeeplyReadItem we got from Ophan and for each
         query CAPI and set the Content for the path.
      */
 
-    ophanApi.getDeeplyReadContent().map { seq =>
+    val deeplyReadFuture = ophanApi.getDeeplyReadContent().flatMap { seq =>
```
Comment (jfsoul) [inline onward/app/feed/DeeplyReadAgent.scala:92]: let's call this `fDeeplyReadItemsWithCapi`

Comment (jfsoul) [reply onward/app/feed/DeeplyReadAgent.scala:92]: and annotate with its type

File: onward/app/feed/DeeplyReadAgent.scala [hunk 2]
```diff
@@ -75,26 +75,23 @@ class DeeplyReadAgent(contentApiClient: ContentApiClient, ophanApi: OphanApi) ex
...
       log.info(s"[cb01a845] ophanItems updated with: ${seq.toArray.size} new items")
-      seq.foreach { ophanItem =>
+      val deeplyReadSeq = seq.map { ophanItem =>
```
Comment (jfsoul) [inline onward/app/feed/DeeplyReadAgent.scala:94]: I think a part or all of this should be extracted into a separate function(s) (not necessarily right now), but calling it `seqCapiFutures` maybe helps for now.

File: onward/app/feed/DeeplyReadAgent.scala [hunk 3]
```diff
@@ -104,51 +101,40 @@ class DeeplyReadAgent(contentApiClient: ContentApiClient, ophanApi: OphanApi) ex
           .showFields("all")
           .showReferences("all")
           .showAtoms("all")
-        val fx = contentApiClient
+        val deeplyReadApiClient = contentApiClient
```
Comment (jfsoul) [inline onward/app/feed/DeeplyReadAgent.scala:104]: let's call this `capiResponse`

File: onward/app/feed/DeeplyReadAgent.scala [hunk 4]
```diff
@@ -104,51 +101,40 @@ class DeeplyReadAgent(contentApiClient: ContentApiClient, ophanApi: OphanApi) ex
           .showFields("all")
           .showReferences("all")
           .showAtoms("all")
```
Comment (jfsoul) [inline onward/app/feed/DeeplyReadAgent.scala:103]: as a quick win maybe we just set `.showReferences("none").showAtoms("none")` or whatever the syntax is

File: onward/app/feed/DeeplyReadAgent.scala (hunk 2 above)
Comment (jfsoul) [reply onward/app/feed/DeeplyReadAgent.scala:94]: and putting the type on it

**Pull request merged**